src/main/java/cubicchunks/server/CubeProviderServer.java

```diff
@@ -197,17 +194,26 @@ public boolean tick() {
         profiler.startSection("providerTick");
         long i = System.currentTimeMillis();
         int randomTickSpeed = this.world.getGameRules().getInt("randomTickSpeed");
+        boolean doMobSpawning = this.world.getGameRules().getBoolean("doMobSpawning");
+        boolean isRaining = this.world.isRaining();
+        boolean isThundering = this.world.isThundering();
         Random rand = this.world.rand;
         PlayerCubeMap playerCubeMap = ((PlayerCubeMap) this.world.getPlayerChunkMap());
         Iterator<Cube> watchersIterator = playerCubeMap.getCubeIterator();
         while (watchersIterator.hasNext()) {
             Cube cube = watchersIterator.next();
             cube.tickCubeServer(() -> System.currentTimeMillis() - i > 40, rand);
-            if (!doRandomBlockTicksHere)
-                continue;
+            this.profiler.startSection("thunder");
+            if (this.world.provider.canDoLightning((Chunk) cube.getColumn()) && isRaining && isThundering && rand.nextInt(100000) == 0)
```

Comment:
Because of this, it might be more efficient to iterate over all columns, do the column checks first and then iterate ove rall cubes in the column. Especially when modded `WorldProviders` are allowed, with arbitrary logic in those methods. Also so the easy checks first (`isRaining` and `isThundering` and randint). I know this is not what vanilla does, but this improvement is essentially free here.

Reply:
You see, vanilla already do it. A problem is when world server checks for lighting strike target it iterate thru all loaded cubes (including not watched by player) and thru all entities in those cubes till it reaches last loaded or world height.